src/graphiql.js

```diff
@@ -0,0 +1,118 @@
+import React, { useRef, useEffect, useState } from 'react';
+import GraphiQL from 'graphiql';
+import PropTypes from 'prop-types';
+import { buildSchema } from 'graphql';
+import GraphiQLExplorer from 'graphiql-explorer';
+import './common/styles/index.scss';
+
+const TykGraphiQL = (props) => {
+  const {
+    fetcher,
+    schema,
+    ideName = 'TykGraphiQL',
+    query: initialQuery = '{}',
+    showPlayground = true,
+    menu,
+  } = props;
+  const playgroundWrapperRef = useRef(null);
+  let playgroundRef = useRef(null);
+  const [explorerIsOpen, setExplorerIsOpen] = useState(false);
+  const [query, setQuery] = useState(initialQuery);
+  const [selectedMenu, setSelectedMenu] = useState(menu?.title);
+
+  useEffect(() => {
+    if (!playgroundWrapperRef.current) {
+      return;
+    }
+
+    playgroundWrapperRef.current.addEventListener('click', (e) => {
```

Comment:
what's the purpose of this ? Just thinking out loud, in case you need to clean the event up

Reply:
the explorer and graphiql are wrapped and tied to the container hence a click on wrapper sometimes trigger the events unintentionally, in order to avoid those this is used.